libs/langgraph/langgraph/pregel/__init__.py

```diff
@@ -432,28 +515,44 @@ def get_state_history(
         for config, checkpoint, metadata, parent_config, _ in self.checkpointer.list(
             config, before=before, limit=limit, filter=filter
         ):
-            with ChannelsManager(
-                self.channels, checkpoint, config
-            ) as channels, ManagedValuesManager(
-                self.managed_values_dict, ensure_config(config), self
-            ) as managed:
-                next_tasks = prepare_next_tasks(
-                    checkpoint,
-                    self.nodes,
-                    channels,
-                    managed,
-                    config,
-                    -1,
-                    for_execution=False,
-                )
-                yield StateSnapshot(
-                    read_channels(channels, self.stream_channels_asis),
-                    tuple(name for name, _ in next_tasks),
-                    config,
-                    metadata,
-                    checkpoint["ts"],
-                    parent_config,
+            # is there a way to do this more efficiently?
+            if include_subgraph_state:
+                checkpoint_tuples = self.checkpointer.list_subgraph_checkpoints(config)
```

Comment:
don't love this as there is a separate list call here, but not sure what's a better way besides somehow combining w/ `.list`

Reply:
but if this looks fine, will implement the same for async and add async tests